host-controller/src/main/java/org/jboss/as/host/controller/DomainModelControllerService.java

```diff
@@ -778,6 +778,8 @@ public ManagementResourceRegistration getRootResourceRegistrationForUpdate(Opera
                     List<ModelNode> domainBootOps = domainPersister.load();
 
                     HostControllerLogger.ROOT_LOGGER.debug("Invoking domain.xml ops");
+                    // https://issues.jboss.org/browse/WFCORE-3897
+                    domainModelComplete.set(true);
```

Comment:
Can you change the name of this field, and the related params and field in RemoteDomainConnectionService, to 'domainConfigAvailable'?  The old 'domainModelComplete' name made sense because of the buggy code I wrote, where it wasn't true until the domain.xml ops had executed and the 'model' (i.e. the internal resource tree) was "complete".  But that's not really what it's used for. It's used to signal to the ModelController that the domain 'config' (i.e. either resource tree or just the ops needed to create it) is 'available' so it can go ahead and validate.